backend/config/curation-api.yml

```diff
@@ -356,7 +356,37 @@ paths:
           $ref: "#/components/responses/401"
         "404":
           $ref: "#/components/responses/404"
-
+  /v1/checks/submission_key:
+    put:
+      summary: Verify the result of submitting a dataset
+      description: >-
+        Check the outcome of submitting a dataset to S3 at the specified key name.
+      operationId: backend.corpora.lambdas.api.v1.curation.check_submission_key.check_submission_key
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              additionalProperties: false
+              properties:
+                s3_object_key:
+                  type: string
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  result:
+                    type: string
+                    enum: ["UPDATE", "NEW", "FAIL"]
```

Comment:
when would it `FAIL`? 🤔

Reply:
Oh, if it doesn't have `.h5ad`...

Reply:
Ah, or if it's to the wrong user prefix, or `/super/` when they don't have permission... right...

Reply:
I skipped checking if the user is the owner or a super. This check is better handled by our S3 credentials and bucket permissions. I assume if the user knows to use super in the title then they must already know if they have it or not. We only check that the use in the key is the owner of the collection. If the user us super then we don't check if they are the owner. 

Also notice that this endpoint does not require Auth. I'm still trying to decide if this is the best course. I figured I'd leave it up to the reviewers and PM.

Reply:
I don't think the user should be passing in the entire s3 key. They should only pass in the collection_id and the identifier, whether tag or dataset_id.